Lib9c/Model/Item/ShopItem.cs

```diff
@@ -49,6 +50,33 @@ public class ShopItem
             Costume = costume;
         }
 
+        public ShopItem(
+            Address sellerAgentAddress,
+            Address sellerAvatarAddress,
+            Guid productId,
+            FungibleAssetValue price,
+            long expiredBlockIndex,
+            INonFungibleItem nonFungibleItem
+        )
+        {
+            SellerAgentAddress = sellerAgentAddress;
+            SellerAvatarAddress = sellerAvatarAddress;
+            ProductId = productId;
+            Price = price;
+            ExpiredBlockIndex = expiredBlockIndex;
```

Comment:
Can `ExpiredBlockIndex` variable be negative? If not, how about checking it and throwing `ArgumentException`?